lib/diagnostic.js

```diff
@@ -10,26 +10,41 @@ const normalizeWords = function (words) {
   let wordArray = words.split(' ');
   const normalize = function (word) {
     return word.replace(/\W+/,'').toUpperCase();
-  }
+  };
 return wordArray.map(normalize);
 };
-
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
-};
+const uniqueWords = function (array) {
+  let result = {};
+  array.forEach(function (word) {
+    result[word] = 0;
+  });
+    //take each value in the array
+  return result;};
 
```

Comment:
Be careful what's this is supposed to take as an arguemtn and what gets returned! You've set this up to take an array, not a string, and returns an object, not an array. It's very close (and actually does the thing), just not in the right js object!